client/flutter/lib/pages/settings_page.dart

```diff
@@ -18,52 +17,71 @@ class SettingsPage extends StatefulWidget {
 }
 
 class _SettingsPageState extends State<SettingsPage> {
-
-  /// TODO: fetch below value from shared pref!
-  bool _switchValue = true;
+  bool _switchValue;
   int indexSelected = 0;
 
+  @override
+  void initState() {
+    super.initState();
+    if (_switchValue == null) {
+      _initSwitch();
+    }
+  }
+
+  _initSwitch() async {
+    _switchValue = await UserPreferences().getAnalyticsEnabled();
+    setState(() {});
+  }
+
+  _toggleAnalytics(bool value) async {
+    await UserPreferences().setAnalyticsEnabled(value);
+  }
+
   @override
   Widget build(BuildContext context) {
-    return ListOfItems([
-      Padding(
-        padding: const EdgeInsets.symmetric(vertical: 15.0, horizontal: 8.0),
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: <Widget>[
-            ListTile(
-              leading: SelectableText(
-                S.of(context).homePagePageSliverListSettingsHeader1,
-                textAlign: TextAlign.start,
-                style: TextStyle(fontSize: 25.0, fontWeight: FontWeight.w700,),
-              ),
-              trailing: CupertinoSwitch(
-                activeColor: Constants.primaryColor,
-                value: _switchValue,
-                onChanged: (value){
-                  /// (1) TODO: updating the stored user preferences
-                  /// (2) TODO: turning Google Analytics on/off
-                  setState(() {
-                    _switchValue = value;
-                  });
-                },
+    return ListOfItems(
+      [
+        Padding(
+          padding: const EdgeInsets.symmetric(vertical: 15.0, horizontal: 8.0),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              ListTile(
+                leading: SelectableText(
+                  S.of(context).homePagePageSliverListSettingsHeader1,
+                  textAlign: TextAlign.start,
+                  style: TextStyle(
+                    fontSize: 25.0,
+                    fontWeight: FontWeight.w700,
+                  ),
+                ),
+                trailing: CupertinoSwitch(
+                  activeColor: Constants.primaryColor,
+                  value: _switchValue ?? false,
+                  onChanged: (newValue) {
+                    _toggleAnalytics(newValue);
+                    setState(() {
```

Comment:
can you make this wait for _toggleAnalytics to finish?  then we show its off only after its off

Reply:
Can do, but it's not required. Also if it's waiting long, button won't change state upon user tap.
Should I change it ?

Reply:
Yes plesae change it